include/seqan3/core/algorithm/all.hpp

```diff
@@ -15,67 +15,9 @@
- * concept. The separate `value` member is used for a proper encapsulation from the actual setting parameter.
- * For example the alignment algorithms require a scoring scheme, but the scoring scheme itself should not be pipeable
- * with other settings.
- *
- * In addition an enum type was defined that will be used later to allow for compatibility checks when combining
- * settings in a configuration object. This enum assigns every configuration element a unique id.
- * To provide compatibility checks the `seqan3::detail::compatibility_table` must be overloaded for the specific
- * algorithm configuration.
- *
- * \include test/snippet/core/algorithm/configuration_compatibility.cpp
- *
- * The type for the configuration element ids is used to overload the bool table. In the example above, both
- * elements can be combined with each other but not with themselves, to avoid inconsistent settings.
- *
- * ### Combining Configurations
- *
- * To enable easy creation of algorithm settings the seqan3::configuration supports a pipeable interface for the
- * different configuration elements which are added through the seqan3::pipeable_config_element base class.
- * The following snippet demonstrates how `bar` and `foo` can be combined.
- *
- * \include test/snippet/core/algorithm/configuration_combine.cpp
-
- * ### Access the data
- *
- * The configuration inherits from a std::tuple and exposes a tuple like interface using the standard
- * position-based and type-based `get` interfaces. The `get` interface was extended to also support
- * template template types as input template parameters to query the correct element:
- *
- * \include test/snippet/core/algorithm/configuration_get.cpp
- *
- * The get interface returns a reference to the stored configuration element. In some cases, e.g. the implementor
- * of the actual algorithm, one wants to have an easy access to the actual value of the setting. Since, the
- * configuration must not contain all possible configuration elements the seqan3::configuration provides a
- * seqan3::configuration::get_or interface. Using this interface one can call get with a specific configuration element.
- * If this configuration element or a specialisation of it is already stored inside of the configuration, the respective
- * element is returned. Otherwise, the passed argument will be returned as the alternative.
- *
- * \include test/snippet/core/algorithm/configuration_get_or.cpp
  */
 #include <seqan3/core/algorithm/algorithm_result_generator_range.hpp>
 #include <seqan3/core/algorithm/bound.hpp>
-#include <seqan3/core/algorithm/concept.hpp>
-#include <seqan3/core/algorithm/configuration_utility.hpp>
-#include <seqan3/core/algorithm/configuration.hpp>
-#include <seqan3/core/detail/pack_algorithm.hpp>
+#include <seqan3/core/algorithm/detail/all.hpp>
 #include <seqan3/core/algorithm/pipeable_config_element.hpp>
+#include <seqan3/core/configuration/all.hpp> // ToDo: remove this with the release SeqAn-3.1.0.
```

Comment:
Not sure how to handle this. This is likely to be forgotten. 
I am thinking of this:
```cpp
#if SEQAN3_VERSION_MAJOR == 3 && SEQAN3_VERSION_MINOR == 1
  #pragma warning "Remove #include <seqan3/core/configuration/all.hpp> from this header."
#endif
```
@marehr what do you think?

Reply:
This was exactly the problem I tried to explain on Gitter on friday..